src/python/pants/engine/target.py

```diff
@@ -2680,16 +2680,21 @@ def rules():
 class InferredDependencies:
     include: FrozenOrderedSet[Address]
     exclude: FrozenOrderedSet[Address]
+    transitive_excludes: FrozenOrderedSet[Address]
```

Comment:
Note I went with `excludes` (plural) to match the other usages in the codebase (instead of singular like `include` and `exclude`)